File: Hour 7/Q2/fibonacci_sequence.cpp
#include<iostream>
#include<math.h>
using namespace std;

bool is_Perfect_Square(int K);
bool is_Fibonacci(int R);

int main(){
    int n, i, count = 0;
    cin>>n;
    int arr[n];
    for(i = 0; i < n; i++)
        cin>>arr[i];
    for (int x : arr) {
        if(is_Fibonacci(x))
            count++;
    }
    cout<<count;
    return 0;
}

bool is_Perfect_Square(int K) {
        int s = (int)sqrt(K);
        return ((s*s == K) ? true: false);
    }

bool is_Fibonacci(int R){
    return(is_Perfect_Square(5*R*R + 4) || is_Perfect_Square(5*R*R - 4));
}

File: Hour 7/Q2/fibonacci_sequence.java
import java.util.Scanner;

public class fibonacci_sequence {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        try{
            int n = sc.nextInt(), i, count = 0;
            int arr[] = new int[n];
            for(i = 0; i < n; i++)
                arr[i] = sc.nextInt();
            for (int x : arr) {
                if(is_Fibonacci(x))
                    count++;
            }
            System.out.print(count);
        }
        finally{
            sc.close();
        }
    }
    
    public static boolean is_Perfect_Square(int K) {
        int s = (int)Math.sqrt(K);
        return ((s*s == K) ? true: false);
    }

    public static boolean is_Fibonacci(int R){
        return(is_Perfect_Square(5*R*R + 4) || is_Perfect_Square(5*R*R - 4));
    }
}
